--- dirmngr-0.5.0/src/crlcache.h.orig	2003-11-11 19:15:01.000000000 +0100
+++ dirmngr-0.5.0/src/crlcache.h	2003-11-17 20:06:25.688540328 +0100
@@ -29,6 +29,9 @@
 #elif HAVE_DB2_DB_H
 #include <limits.h>
 #include <db2/db.h>
+#elif HAVE_DB_H
+#include <limits.h>
+#include <db.h>
 #else
 /* Error */
 #endif
--- dirmngr-0.5.0/src/crlcache.c.orig	2003-11-17 11:58:04.000000000 +0100
+++ dirmngr-0.5.0/src/crlcache.c	2003-11-17 20:07:30.708655768 +0100
@@ -87,13 +87,17 @@
 static int 
 mydbopen (const char* file, DB** dbpp)
 {
-#ifdef HAVE_DB4_DB_H
+#if (DB_VERSION_MAJOR >= 4)
   int rc = db_create( dbpp, 0, 0 );
   if( !rc ) {
+#if (DB_VERSION_MINOR >= 1)
+    rc = (*dbpp)->open( *dbpp, NULL, file, 0, DB_HASH, DB_CREATE, S_IRUSR|S_IWUSR);
+#else
     rc = (*dbpp)->open( *dbpp, file, 0, DB_HASH, DB_CREATE, S_IRUSR|S_IWUSR);
+#endif
   }
   return rc;
-#elif HAVE_DB2_DB_H
+#elif (DB_VERSION_MAJOR >= 2)
   return db_open( file, DB_HASH, DB_CREATE, S_IRUSR|S_IWUSR, NULL, NULL, dbpp );
 #endif
 }
@@ -1038,7 +1042,7 @@
   fprintf (fp, " Issuer Hash:\t%s\n", e->issuer_hash );
   fprintf (fp, " Valid from %s to %s\n\n", e->this_update, e->next_update); 
 
-#if defined(HAVE_DB4_DB_H) || defined(DB2_CURSOR_4_PARAMS)
+#if (DB_VERSION_MAJOR >= 4) || defined(DB2_CURSOR_4_PARAMS)
 #define X ,0
 #endif
   rc = e->dbf->cursor( e->dbf, NULL, &cursorp X);
--- dirmngr-0.5.0/configure.ac.orig	2003-11-17 11:58:04.000000000 +0100
+++ dirmngr-0.5.0/configure.ac	2003-11-17 20:06:25.693539568 +0100
@@ -288,7 +288,7 @@
 AC_HEADER_STDC
 AC_CHECK_HEADERS([string.h])
 ac_have_db4=yes
-AC_CHECK_HEADERS(db4/db.h,,[ac_have_db4=no])
+AC_CHECK_HEADERS(db4/db.h,,[AC_CHECK_HEADERS(db.h,,[ac_have_db4=no])])
 AC_CHECK_LIB(db-4,main,,[ac_have_db4=no])
 
 if test "$ac_have_db4" = "no" ; then 
